sdk/azidentity/client_secret_credential.go

```diff
@@ -54,47 +55,62 @@ func (c *ClientSecretCredential) AuthenticationPolicy(options azcore.Authenticat
 }
 
 type bearerTokenPolicy struct {
-	// take lock when manipulating header/expiresOn fields
-	lock      sync.RWMutex
-	header    string
-	expiresOn time.Time
+	updating  int64                  // atomically set to 1 to indicate a refresh is in progress
+	header    atomic.Value           // string
```

Comment:
Because atomic stuff is so error prone and it's easy for someone to assert incorrectly or just access the field directly without using atomic.GetValue, could we add some some structs with members like what Uber has done here: https://github.com/uber-go/atomic/blob/master/atomic.go.

We just need to add the types/methods we require and it can grow over time.

Reply:
Fixed.